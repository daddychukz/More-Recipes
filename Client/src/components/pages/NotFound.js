import React from 'react';
import { Header } from './Header';
import { Footer } from './Footer';

const NotFound = () => {
    return (
        <div>
            <Header />
            
            <header id="header">
            
            </header>
            
            <section>
                <div className='container'>
                    <div className='row'>
                        <div className='col-6 obj-color'>
                            <h2>404! Resource Not Found</h2>
                        </div>
                    </div>
                </div>
            </section>
        </div>
    );
};

export default NotFound;

