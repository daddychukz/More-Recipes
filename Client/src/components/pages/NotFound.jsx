import React from 'react';
import Header from './Header';

const NotFound = () => (
  <div>
    <Header />

    <header id="header" />

    <section>
      <div className="container">
        <div className="row">
          <div className="col-6 obj-color">
            <h2>404! Resource Not Found</h2>
          </div>
        </div>
      </div>
    </section>
  </div>
);

export default NotFound;
