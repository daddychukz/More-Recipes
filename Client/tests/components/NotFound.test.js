import React from 'react';
import { mount } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import render from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import NotFound from '../../src/components/pages/NotFound';


const middleware = [thunk];
const mockStore = configureMockStore(middleware);
const initialState = {
  auth: {
    isAuthenticated: false,
    user: {}
  },
  user: {},
  recipeOperations: {
    pagination: {},
    recipes: []
  },
  review: [],
  favorite: [],
  popularRecipes: [],
  singleRecipe: {},
  allRecipe: {
    pagination: {},
    recipes: []
  }
};

const store = mockStore(initialState);

const testRender = render.create(
  <BrowserRouter>
    <Provider store={store}>
      <NotFound />
    </Provider>
  </BrowserRouter>
);
describe('Not Found page test suite', () => {
  const wrapper = mount(
    <BrowserRouter>
      <Provider store={store}>
        <NotFound />
      </Provider>
    </BrowserRouter>
  );
  it('renders NotFound component', () => {
    expect(testRender).toMatchSnapshot();
  });

  it('should check the <Header/> component exists', () => {
    expect(wrapper.find('Header').length).toEqual(1);
  });
});
