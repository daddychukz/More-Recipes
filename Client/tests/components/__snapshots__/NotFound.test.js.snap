// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not Found page test suite renders NotFound component 1`] = `
<div>
  <div>
    <nav
      className="navbar navbar-expand-md bg-dark navbar-dark fixed-top"
    >
      <div
        className="container"
      >
        <a
          className="navbar-brand"
          href="/recipe-box"
          onClick={[Function]}
        >
          <h3
            id="logo"
          >
            More-Recipes
          </h3>
        </a>
        <button
          className="navbar-toggler"
          data-target="#navbarCollapse"
          data-toggle="collapse"
        >
          <span
            className="navbar-toggler-icon"
          />
        </button>
        <div
          className="collapse navbar-collapse"
          id="navbarCollapse"
        >
          <ul
            className="navbar-nav ml-auto"
          >
            <li
              className="nav-item"
            >
              <a
                className="nav-link false"
                href="/recipe-box"
                id="home"
                onClick={[Function]}
              >
                Home
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/add-recipe"
                id="addRecipe"
                onClick={[Function]}
              >
                Add Recipe
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/my-recipe"
                id="my-recipe"
                onClick={[Function]}
              >
                My Recipes
              </a>
            </li>
            <li
              className="nav-item"
            >
              <a
                className="nav-link"
                href="/my-favorite"
                onClick={[Function]}
              >
                Favorites
              </a>
            </li>
            <li
              className="nav-item"
            >
              <div
                className="btn-group open"
              >
                <i
                  aria-hidden="true"
                  className="fa fa-user-circle fa-2x pull-right"
                  data-toggle="dropdown"
                  id="drop-down"
                />
                <ul
                  className="dropdown-menu"
                >
                  <li>
                    <a
                      className="dropdown-item"
                      href="/my-profile"
                      onClick={[Function]}
                    >
                      Profile
                    </a>
                  </li>
                  <li
                    className="divider"
                  />
                  <li>
                    <a
                      className="dropdown-item"
                      href="/"
                      id="logout"
                      onClick={[Function]}
                    >
                      Logout
                    </a>
                  </li>
                </ul>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      className="modal fade text-dark"
      id="contactModal"
    >
      <div
        className="modal-dialog"
      >
        <div
          className="modal-content"
        >
          <div
            className="modal-header bg-primary"
          >
            <h5
              className="modal-title"
              id="contactModalTitle"
              style={
                Object {
                  "color": "white",
                }
              }
            >
              Contact Us
            </h5>
          </div>
          <div
            className="modal-body"
          >
            <form>
              <div
                className="form-group"
              >
                <label
                  htmlFor="name"
                >
                  Name
                </label>
                <input
                  className="form-control"
                  type="text"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="email"
                >
                  Email
                </label>
                <input
                  className="form-control"
                  type="email"
                />
              </div>
              <div
                className="form-group"
              >
                <label
                  htmlFor="messager"
                >
                  Message
                </label>
                <textarea
                  className="form-control"
                />
              </div>
            </form>
          </div>
          <div
            className="modal-footer"
          >
            <button
              className="btn btn-primary btn-block"
            >
              Submit
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
  <header
    id="header"
  />
  <section>
    <div
      className="container"
    >
      <div
        className="row"
      >
        <div
          className="col-6 obj-color"
        >
          <h2>
            404! Resource Not Found
          </h2>
        </div>
      </div>
    </div>
  </section>
</div>
`;
